src/main/java/com/mraof/minestuck/client/model/UnderlingModel.java

```diff
@@ -0,0 +1,29 @@
+package com.mraof.minestuck.client.model;
+
+import com.mraof.minestuck.Minestuck;
+import com.mraof.minestuck.entity.underling.UnderlingEntity;
+import net.minecraft.util.ResourceLocation;
+import software.bernie.geckolib3.model.AnimatedGeoModel;
+
+import java.util.Objects;
+
+public class UnderlingModel<T extends UnderlingEntity> extends AnimatedGeoModel<T> {
+    @Override
+    public ResourceLocation getModelLocation(T entity) {
+        return new ResourceLocation(Minestuck.MOD_ID, "geo/entity/underlings/" + getName(entity) + ".geo.json");
+    }
```

Comment:
Lacking a better place to put it, I will be informing Riotmode that the leg bones of Imps Z fight at what would be the knee. This is not a fix request or TODO

Reply:
Other such animation requests will be made for a different PR potentially even with a different author, as this is just the framework and not the fine tweaking of these systems

Reply:
ill leave the comments unresolved until we have a complete list :D